ignite/handlers/checkpoint.py

```diff
@@ -286,15 +287,17 @@ def __init__(
             if "checkpointer" in to_save:
                 raise ValueError(f"Cannot have key 'checkpointer' if `include_self` is True: {to_save}")
 
-        if not (callable(save_handler) or isinstance(save_handler, BaseSaveHandler)):
-            raise TypeError("Argument `save_handler` should be callable or inherit from BaseSaveHandler")
+        if not (isinstance(save_handler, str) or callable(save_handler) or isinstance(save_handler, BaseSaveHandler)):
+            raise TypeError("Argument `save_handler` should be a string or callable or inherit from BaseSaveHandler")
 
         if global_step_transform is not None and not callable(global_step_transform):
             raise TypeError(f"global_step_transform should be a function, got {type(global_step_transform)} instead.")
 
         self.to_save = to_save
         self.filename_prefix = filename_prefix
         self.save_handler = save_handler
+        if isinstance(self.save_handler, str):
+            self.save_handler = DiskSaver(self.save_handler, create_dir=True)
```

Comment:
mypy is not happy with this type redefinition. Let's recode that as 
```python
if isinstance(save_handler, str):
    self.save_handler = DiskSaver(save_handler, create_dir=True)
else:
    self.save_handler = save_handler
```
Please check locally if this helps to solve mypy issue